tests/python_package_test/test_sklearn.py

```diff
@@ -174,17 +178,19 @@ def test_sklearn_backward_compatibility(self):
         np.testing.assert_allclose(y_pred_1, y_pred_2)
 
     def test_sklearn_integration(self):
-        # we cannot use `check_estimator` directly since there is no skip test mechanism
-        for name, estimator in ((lgb.sklearn.LGBMClassifier.__name__, lgb.sklearn.LGBMClassifier),
-                                (lgb.sklearn.LGBMRegressor.__name__, lgb.sklearn.LGBMRegressor)):
-            check_parameters_default_constructible(name, estimator)
-            check_no_fit_attributes_set_in_init(name, estimator)
-            # we cannot leave default params (see https://github.com/Microsoft/LightGBM/issues/833)
-            estimator = estimator(min_data=1, min_data_in_bin=1)
-            for check in _yield_all_checks(name, estimator):
-                if check.__name__ == 'check_estimators_nan_inf':
-                    continue  # skip test because LightGBM deals with nan
-                try:
-                    check(name, estimator)
-                except SkipTest as message:
-                    warnings.warn(message, SkipTestWarning)
+        # sklearn <0.19 cannot accept instance, but many tests could be passed only with min_data=1 and min_data_in_bin=1
+        if sklearn_at_least_019:
+            # we cannot use `check_estimator` directly since there is no skip test mechanism
+            for name, estimator in ((lgb.sklearn.LGBMClassifier.__name__, lgb.sklearn.LGBMClassifier),
+                                    (lgb.sklearn.LGBMRegressor.__name__, lgb.sklearn.LGBMRegressor)):
+                check_parameters_default_constructible(name, estimator)
+                check_no_fit_attributes_set_in_init(name, estimator)
+                # we cannot leave default params (see https://github.com/Microsoft/LightGBM/issues/833)
+                estimator = estimator(min_data=1, min_data_in_bin=1)
```

Comment:
Cn you try`min_child_samples=1', not `min_data=1`?

Reply:
in other words, I don't know what happens when we set both `min_data` and `min_child_samples` (the latter is one of parameters of scikit-learn interface).

Reply:
@henry0312 
Please provide the full line `estimator = estimator(...) ` which I should write for the test.

As I understand, parameters from `params` have higher priority and overrides parameters of scikit-learn interface (https://github.com/Microsoft/LightGBM/blob/master/python-package/lightgbm/sklearn.py#L271).

Reply:
`estimator(min_child_samples=1, min_data_in_bin=1)` is ok for now.